tap_azure_git/__init__.py

```diff
@@ -710,37 +712,48 @@ def get_all_commit_files(schemas, org, repo_path, state, mdata, start_date, gitL
-                annotated_tag = gitLocal.getAnnotatedTag(gitLocalRepoPath, ref['sha'])
-                headsToCommits[h] = {
-                    'type': 'tag',
-                    'sha': annotated_tag['tagId'],
-                    'commit_sha': annotated_tag['commitId']
-                }
-
-                if 'annotated_tags' in schemas:
-                    annotated_tag_record = {
-                        **annotated_tag,
-                        '_sdc_repository': sdcRepository,
-                        'id': '{}/{}'.format(sdcRepository, annotated_tag['tagId'])
+    # Skip annotated tags processing in commits_only mode
+    if not commits_only:
+        with metrics.record_counter('annotated_tags') as counter:
+            for h in localHeads:
+                ref = localHeads[h]
+                if ref['type'] == 'tag':
+                    annotated_tag = gitLocal.getAnnotatedTag(gitLocalRepoPath, ref['sha'])
+                    headsToCommits[h] = {
+                        'type': 'tag',
+                        'sha': annotated_tag['tagId'],
+                        'commit_sha': annotated_tag['commitId']
                     }
-                    annotated_tag_record['tagger']['date'] = \
-                        annotated_tag_record['tagger']['date'].isoformat()
-                    with singer.Transformer() as transformer:
-                        rec = transformer.transform(annotated_tag_record, schemas['annotated_tags'],
-                            metadata=metadata.to_map(mdata))
-                    counter.increment()
-                    singer.write_record('annotated_tags', rec, time_extracted=extraction_time)
 
-            else:
-                headsToCommits[h] = {
-                    'type': 'commit',
-                    'sha': ref['sha'],
-                    'commit_sha': ref['sha']
-                }
+                    if 'annotated_tags' in schemas:
+                        annotated_tag_record = {
+                            **annotated_tag,
+                            '_sdc_repository': sdcRepository,
+                            'id': '{}/{}'.format(sdcRepository, annotated_tag['tagId'])
+                        }
+                        annotated_tag_record['tagger']['date'] = \
+                            annotated_tag_record['tagger']['date'].isoformat()
+                        with singer.Transformer() as transformer:
+                            rec = transformer.transform(annotated_tag_record, schemas['annotated_tags'],
+                                metadata=metadata.to_map(mdata))
+                        counter.increment()
+                        singer.write_record('annotated_tags', rec, time_extracted=extraction_time)
+
+                else:
+                    headsToCommits[h] = {
+                        'type': 'commit',
+                        'sha': ref['sha'],
+                        'commit_sha': ref['sha']
+                    }
+    else:
```

Comment:
I don't understand why we are making this change.  If we just add each reference from localHeads as a commit, we will potentially output refs incorrectly.  The refs which are `annotated_tags` will then be stored as `commits`.

When we iterate over the references, the code is already prepared to navigate through commits and annotated tags.  To fetch all the `commit_files_meta` or `commit_files` we should  comprehend `annotated_tags` as we are.  `getAnnotatedTag` should be very fast and the annotated tags should be available, even when using `--filter=tree:0`